model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/authorization/StoreFactoryCacheSession.java

```diff
@@ -672,7 +672,17 @@ public Resource findByName(String name, String ownerId, String resourceServerId)
         @Override
         public void findByOwner(String ownerId, String resourceServerId, Consumer<Resource> consumer) {
             String cacheKey = getResourceByOwnerCacheKey(ownerId, resourceServerId);
-            cacheQuery(cacheKey, ResourceListQuery.class, () -> getResourceStoreDelegate().findByOwner(ownerId, resourceServerId),
+            cacheQuery(cacheKey, ResourceListQuery.class, () -> {
+                        List<Resource> resources = new ArrayList<>();
+                        getResourceStoreDelegate().findByOwner(ownerId, resourceServerId, new Consumer<Resource>() {
+                            @Override
+                            public void accept(Resource resource) {
+                                resources.add(resource);
+                                consumer.accept(resource);
+                            }
+                        });
+                        return resources;
+                    },
                     (revision, resources) -> new ResourceListQuery(revision, cacheKey, resources.stream().map(resource -> resource.getId()).collect(Collectors.toSet()), resourceServerId), resourceServerId, consumer);
```

Comment:
```suggestion
                    (revision, resources) -> new ResourceListQuery(revision, cacheKey, resources.stream().map(Resource::getId).collect(Collectors.toSet()), resourceServerId), resourceServerId, consumer);
```

Reply:
@pedroigor there are lot of places within `StoreFactoryCacheSession.java` where lambda function can be replaced with static reference. I would say it makes sense to replace all occurrences. WDYT?

Reply:
Sure